### ISSUE TITLE
Simple compiler bug

### ISSUE THREAD

**Issue by deltamolfar** [wiremod/wire#3130]
**Describe the bug**
Compiler bug

**To Reproduce**
Attempt to call nil lambda.
```
@persist A:function

A()
```

**Expected behavior**
Verbose error

**Error message**
`sv: Expression 2 (generic): Internal error 'addons/wire/lua/entities/gmod_wire_expression2/base/compiler.lua:1884: attempt to index local 'f' (a nil value)' at line 3, char 1`

**Comment by Vurv78** [comment]
It is intentional to avoid overhead with the nil check since the error is harmless. That would ideally be a compile error since the variable isn't initialized though.

No functions should give you a nil lambda

**Comment by Fasteroid** [comment]
a more user-friendly error might be nice though

**Comment by Zvbhrf** [comment]
Hello, 
today I've encountered this issue, any updates on it
?

**Comment by Vurv78** [comment]
> Hello, today I've encountered this issue, any updates on it ?

No. As I said, it's intentional, regardless you'd get an error message. Only difference is that the message or being at compile time could change.

I'm not going to do it because I'm not going to associate with Wiremod as an organization that defends weird pedophiles in a game largely inhabited by children.

That is if you even see this message as they'll probably delete it and ban me from the organization at some point anyway

**Comment by thegrb93** [comment]
Your own sus activities is why you didn't get back on the team, if you even cared about it. Your self righteous acting and taking it out on them is insanely delusional.

**Comment by Vurv78** [comment]
> Your own sus activities is why you didn't get back on the team, if you even cared about it. Your self righteous acting and taking it out on them is insanely delusional.

This is the pedophile in question for those reading..

I never wanted "back in the team" because I don't want to associate with you weirdos.

As for "sus activities": I think messing with servers that have Starfall on them doesn't compare to the danger you pose with authority in a game composed of minors.

But I will admit it was immature, sure. I can admit faults rather than being a weird sex pest and dying on the hill of not wanting to remove porn from the default StarfallEx whitelist (still largely a children's game, btw) like you.

PS: Wanna stop blatantly stealing my Neostarfall code and obfuscating it in SFEx? :v)

**Comment by thegrb93** [comment]
Don't call me a pedo, bud. And just because my addon allows people to use images from anywhere on the internet doesn't constitute that.

**Comment by Fasteroid** [comment]
@Anticept might wanna lock this one, it's getting spicy in here

**Comment by Anticept** [comment]
I will deal with you two tomorrow.

**Comment by Anticept** [comment]
@thevurv You were trying to be reached by fasteroid and i on discord a few weeks ago but couldn't get through. Then I remembered this topic. The bans have all been lifted a while ago. You can do what you want with that, join or not, just remember the condition I asked in the convo if you do.
